pkg/setup/initialize.go

```diff
@@ -123,14 +124,31 @@ func initialize(
 				klog.Fatalf("Error running k3s: %v", err)
 			}
 		}()
-	} else if certificatesDir != "" {
-		// generate k8s certificates
-		err = GenerateK8sCerts(ctx, currentNamespaceClient, vClusterName, currentNamespace, serviceCIDR, certificatesDir, options.ClusterDomain)
-		if err != nil {
-			return err
+	case constants.K8SDistro, constants.EKSDistro:
+		if certificatesDir != "" {
+			// generate k8s certificates
+			err = GenerateK8sCerts(ctx, currentNamespaceClient, vClusterName, currentNamespace, serviceCIDR, certificatesDir, options.ClusterDomain)
+			if err != nil {
+				return err
+			}
+			apiUp := make(chan struct{})
+			// start k8s
+			go func() {
+				// we need to run this with the parent ctx as otherwise this context will be cancelled by the wait
+				// loop in Initialize
+
+				err := k8s.StartK8S(parentCtx, apiUp)
+
+				if err != nil {
+					klog.Fatalf("Error running k8s: %v", err)
+				}
+			}()
+			klog.Info("waiting for the api to be up")
+			<-apiUp
```

Comment:
Why do we need to wait here? We also don't wait for the other distros do we?

Reply:
yes, otherwise the controller crashes, and then we get error outputs on the cli. Since we know this is happening at the start, we should wait for it